### PR TITLE
Update cpp pointer docs

### PR DESCRIPTION
**Pull request by arinze19**
### Type of Change

- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:18]:
```suggestion
A pointer variable is declared using the indirection operator (`*`). This is a special unary operator that signals the program to store the memory address of another variable.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:20]:
```suggestion
To get the memory address of a variable to store in the pointer variable, the address operator (`&`) is used. This signals the program to return the memory address of a variable. Memory addresses are usually prefixed with `0x` this just indicates that the following digits are in hexadecimal notation.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:37]:
```suggestion
There aren't many of useful operations that can be performed using just the memory address that pointers store. Instead, there needs to be a way to get the value stored in the memory address so it can be manipulated directly.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:38]:
```suggestion
Dereferencing a pointer helps to do this. A pointer is derefrenced with the indirection operator (`*`) as well.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:39]:
```suggestion
When a pointer is dereferenced, it tells the program to return the value contained in the specified memory address and then operations can be performed on it.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:61]:
```suggestion
This however is a cause of concern because if the program tries to dereference and manipulate a pointer that holds no memory address, it could lead to some unexpected behavior. To prevent this, `nullptr` is used, which is a special kind of value that prevents our pointers from behaving unexpectedly when they hold no memory address.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (SSwiniarski) [inline content/cpp/concepts/pointers/pointers.md:78]:
```suggestion
Pointers can also be passed into functions as arguments. This helps to manipulate values from within a function even though the value lives outside that function
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arinze19 I did a first review and had a number of line edits because we avoid first- and second-person pronouns in Docs.

**Comment by arinze19** [comment]
> @arinze19 I did a first review and had a number of line edits because we avoid first- and second-person pronouns in Docs.

Brilliant idea 👍

**Review by SSwiniarski — APPROVED** [review]
@arinze19 I went ahead and committed my suggestions myself so we can get this moving to a second review.

File: content/cpp/concepts/pointers/pointers.md
Comment (KTom101) [inline content/cpp/concepts/pointers/pointers.md:38]:
```suggestion
This can be done by dereferencing a pointer. The indirection operator (`*`) used for declaring a pointer can be used to dereference a pointer as well.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (KTom101) [inline content/cpp/concepts/pointers/pointers.md:53]:
```suggestion
Similar to other data types, pointers can be declared without being assigned a value.
```

File: content/cpp/concepts/pointers/pointers.md
Comment (KTom101) [inline content/cpp/concepts/pointers/pointers.md:83]:
```suggestion
  *game = new_game; // Dereference the pointer and change the game
```

File: content/cpp/concepts/pointers/pointers.md
Comment (KTom101) [inline content/cpp/concepts/pointers/pointers.md:87]:
```suggestion
changeGame(&game, "Baseball"); // Passing a pointer as a function argument
```

File: content/cpp/concepts/pointers/pointers.md
Comment (KTom101) [inline content/cpp/concepts/pointers/pointers.md:76]: AP-style title case
```suggestion
## Pointers As Function Parameters
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@arinze19 I added a few changes to this.

File: content/cpp/concepts/pointers/pointers.md
Comment (yangc95) [inline content/cpp/concepts/pointers/pointers.md:85]:
```suggestion
```

**Comment by yangc95** [comment]
@arinze19 due to the time that has elapsed, I went ahead and made changes as per @KTom101 's suggestions 👍 please let me know if there are any concerns

**Comment by yangc95** [comment]
> Hey 👋 Sorry I haven’t been able to follow up on this been quite busy. Looks good to me

No worries! Awaiting approval for this one from one of us maintainers, but it LGTM otherwise ^^ Hope all is well with your studies

**Review by KTom101 — APPROVED** [review]
@yangc95 @arinze19 LGTM 👍

**Pull request merged**
